test/node/msw-api/setup-server/scenarios/custom-interceptors.node.test.ts

```diff
@@ -0,0 +1,37 @@
+import { HttpResponse, http } from 'msw'
+import { SetupServerApi } from 'msw/node'
+import { FetchInterceptor } from '@mswjs/interceptors/fetch'
+import { XMLHttpRequestInterceptor } from '@mswjs/interceptors/XMLHttpRequest'
+
+const fetchInterceptorSpy = vi.spyOn(FetchInterceptor.prototype, 'apply')
```

Comment:
Let's drop these spies and instead:

1. Declare a request handler as you normally would.
2. Assert that the matching request receives the mocked response. 

This will give us assurance that the custom interceptor indeed works.